### ISSUE TITLE
 Allow copying an existing question

### ISSUE THREAD

**Issue by ipeevski** [metabase/metabase#4483]
I know it's possible to do this by opening a question, then changing a parameter, then saving as a new question. 

While that functionality is perfect and completely satisfactory, it was very confusing for me to find it, so it might be worth duplicating the functionality under "edit" an existing question. So maybe have a duplicate / save as copy icon at the top (next to the save changes)

⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by dawoodjee** [comment]
Yes please. This question, collention, and dashboard **copy option** would be great to also have in list view.

**Comment by camsaul** [comment]
Hey everyone, one way we prioritize issues is by sorting them by the number of 👍 reactions on each issue's description. So instead of leaving `+1` or 👍  comments, please just upvote by adding a reaction to the issue description.

**Comment by MarkRx** [comment]
## Implementation Details
- A copy of a dashboard will duplicate all filters and cards including card settings overrides
- A copy of a dashboard will not duplicate questions - all cards will reference the same underlying option
- (stretch) an option when copying a dashboard could be added to do a "deep copy" of questions
- The creator of the copied dashboard will be the user that initiated the copy
- The create / update timestamps of the new dashboard will be the time the copy was made (though I can see a case where those would remain the same as that is an approach software sometimes takes)
- A dashboard can only be copied if the user can read the source dashboard and can write to the target collection. A copy should not be prevented if the dashboard contains cards that the user cannot view - the new dashboard would simply also have a corresponding card that cannot be viewed.

## UI
A "Copy this item" dropdown should be added to the collections page:
![collection_single_copy_dropdown](https://user-images.githubusercontent.com/17504314/47045554-14745000-d161-11e8-99c3-b82a3cc57beb.PNG)

Copying a single dashboard will open up a dialog with a default of <name> - Copy, <description>, <current collection>
![copy_dashboard_modal](https://user-images.githubusercontent.com/17504314/47045609-3a99f000-d161-11e8-95dd-c139cb531923.PNG)

Copying multiple dashboards at once should also be supported. It should reuse the same dialog as moving multiple items. All copies will have - Copy appended to their names.
![dashboard_multi_copy_dialog](https://user-images.githubusercontent.com/17504314/47045625-44bbee80-d161-11e8-8f22-798d769fad8e.PNG)
![dashboard_multi_copy](https://user-images.githubusercontent.com/17504314/47045651-530a0a80-d161-11e8-8265-b735c4519fbd.PNG)

Copying a dashboard should also be available from the dashboard itself.
![dashboard_copy_from_dashboard](https://user-images.githubusercontent.com/17504314/47045679-64531700-d161-11e8-8b24-d26c0124e263.PNG)

**Comment by vvaezian** [comment]
With [this](https://github.com/vvaezian/metabase_api_python) Python wrapper for Metabase API you can copy cards, dashboards (with the option to deepcopy), pulses and collections (with the option to deepcopy the child dashboards).
```python
from metabase_api import Metabase_API

mb = Metabase_API('https://...', 'username', 'password')
mb.copy_card(source_card_name='test_card', destination_collection_id=123)
```

**Comment by daltojohnso** [comment]
@mazameli / @kdoh -- I took an initial pass at this. Put the action inside the pencil icon menu. The modal is identical to the Save modal except that I've made the title "Copy question." LMK what you think & what changes you want made to it.

![Screen Shot 2021-02-05 at 2 08 39 PM](https://user-images.githubusercontent.com/13057258/107094443-d8df7300-67bb-11eb-94e0-dd5a1126cb9e.png)

![Screen Shot 2021-02-05 at 2 09 00 PM](https://user-images.githubusercontent.com/13057258/107094446-db41cd00-67bb-11eb-811e-dfcbb0bd19e9.png)
